src/Endatix.Framework/Platform/AssemblyExtensions.cs

```diff
@@ -0,0 +1,56 @@
+using System.Reflection;
+
+namespace Endatix.Framework;
+
+/// <summary>
+/// Extension methods for the Assembly class
+/// </summary>
+public static class AssemblyExtensions
+{
+    /// <summary>
+    /// Get all Endatix platform assemblies
+    /// </summary>
+    /// <param name="assembly">The assembly to start from</param>
+    /// <returns>A HashSet of Endatix platform assemblies to ensure no duplication of assemblies</returns>
+    public static HashSet<Assembly> GetEndatixPlatformAssemblies(this Assembly assembly)
+    {
+        HashSet<Assembly> endatixAssemblies = [];
+        if (assembly == null || !assembly.IsEndatixAssembly())
```

Comment:
I don't understand the purpose of this check. Firstly, do we expect `GetEndatixPlatformAssemblies` to be called on `null`, as it is an extension method?
The second part is also not quite clear to me - we don't wish to continue if the current assembly is not an Endatix assembly even if it's another product that references Endatix?